lib/gems/pending/util/postgres_admin.rb

```diff
@@ -136,12 +136,25 @@ def self.backup_pg_compress(opts)
     opts.delete(:dbname)
 
     path = Pathname.new(opts.delete(:local_file))
```

Comment:
Now, this thing won't always be a path right? Eventually you'll be passing in some other IO object here so you can do splitting, right?

I guess you'll have to come in and make changes here at that point anyway ...

Reply:
No, everything is actually a path so far in https://github.com/ManageIQ/manageiq-gems-pending/pull/361

That made things that way since they were easier to pass to commands in the way they already were being used (i.e. the `:local_file` option in this class).

Unfortunately, I didn't forsee having to do make the change in this PR, so IO objects may have been a more graceful way to handle things in the long run, but oh well...

Reply:
I think I did a poor job of explaining that... let me try that again...

Since we were using `AwesomeSpawn` before, we had zero access to the `:in` and `:out` streams of the command we were passing, so this lib always passed file paths to `pg_*` executables we were shelling out to.

While I did switch to `Kernel.spawn` here so I had access to those streams to make this PR function (do to the limitations of `pg_basebackup` and `--pgdata`), I kept the interface the same as it was before, which assumed file paths, which is how I designed things in #361 as well (so file splitting works fine, and we just read from the `fifo`).

Reply:
Okay, so this will just be a named pipe which still qualifies as a file path :ok_hand: Got it